include/aws/io/retry_strategy.h

```diff
@@ -106,14 +106,16 @@ enum aws_exponential_backoff_jitter_mode {
  * "use defaults"
  */
 struct aws_exponential_backoff_retry_options {
-    /** Event loop group to use for scheduling tasks. */
+    /* Event loop group to use for scheduling tasks. */
     struct aws_event_loop_group *el_group;
-    /** Max retries to allow. The default value is 10 */
+    /* Max retries to allow. The default value is 10 */
     size_t max_retries;
-    /** Scaling factor to add for the backoff. Default is 500ms */
+    /* Scaling factor to add for the backoff. Default is 500ms */
     uint32_t backoff_scale_factor_ms;
-    /** Max retry backoff in seconds. Default is 20 seconds */
+    /* Max retry backoff in seconds. Default is 20 seconds */
     uint32_t max_backoff_secs;
+    /* Disables retries completely. Attempts to acquire_token will fail with `AWS_IO_RETRY_PERMISSION_DENIED` */
+    bool no_retries;
```

Comment:
I am not a fan of the no_retries config hidden inside of the backoff retry options.

I'd suggest to create a new API with `aws_retry_strategy_new_no_retry()` to be more clear.

Reply:
Thanks, I thought it adds to much boiler late for a simple case of no_retries but it makes sense to have it as a separate strategy. I have updated it.